packages/@aws-cdk/aws-s3/lib/perms.ts

```diff
@@ -28,3 +28,7 @@ export const KEY_WRITE_ACTIONS = [
   'kms:ReEncrypt*',
   'kms:GenerateDataKey*',
 ];
+
+export const KEY_WRITE_ACTIONS_FOR_MULTIPART = [
+  'kms:Decrypt', // required for multipart uploads. Refer https://aws.amazon.com/premiumsupport/knowledge-center/s3-multipart-kms-decrypt/
```

Comment:
Why not add this to the `KEY_WRITE_ACTIONS` array? Would make it apply across the board for all write operations.